lib/async_hooks.js

```diff
@@ -255,18 +255,15 @@ class AsyncLocalStorage {
     resource[this.kResourceStore] = store;
   }
 
-  runSyncAndReturn(store, callback, ...args) {
-    const resource = executionAsyncResource();
-    const outerStore = resource[this.kResourceStore];
-    this.enterWith(store);
-    try {
+  run(store, callback, ...args) {
+    const resource = new AsyncResource('AsyncLocalStorage');
```

Comment:
Does it makes sense to name the resource `'AsyncLocalStorage.run'`? It would separate the ones created for `run` and the ones created for `exit`

Reply:
I'm good with whatever name everyone thinks is reasonable. What do you think, @puzpuzpuz?

Reply:
Considering the fact that `AsyncResource` is only instantiated here, we could leave it as is, but I'm also fine with both options.